FreeRTOS_ARP.c

```diff
@@ -308,6 +311,29 @@ static void vARPProcessPacketReply( ARPPacket_t * pxARPFrame,
 
     iptracePROCESSING_RECEIVED_ARP_REPLY( ulTargetProtocolAddress );
     vARPRefreshCacheEntry( &( pxARPHeader->xSenderHardwareAddress ), ulSenderProtocolAddress, pxTargetEndPoint );
+
+    if( pxARPWaitingNetworkBuffer != NULL )
+    {
+        IPPacket_t * pxARPWaitingIPPacket = ipCAST_PTR_TO_TYPE_PTR( IPPacket_t, pxARPWaitingNetworkBuffer->pucEthernetBuffer );
+        IPHeader_t * pxIPHeader = &( pxARPWaitingIPPacket->xIPHeader );
+
+        if( ulSenderProtocolAddress == pxIPHeader->ulSourceIPAddress )
+        {
+            IPStackEvent_t xEventMessage;
+            const TickType_t xDontBlock = ( TickType_t ) 0;
+
+            xEventMessage.eEventType = eNetworkRxEvent;
+            xEventMessage.pvData = ( void * ) pxARPWaitingNetworkBuffer;
+
+            xSendEventStructToIPTask( &xEventMessage, xDontBlock );
```

Comment:
Oops, the return value should be checked:
~~~c
-    xSendEventStructToIPTask( &xEventMessage, xDontBlock )
+    if( xSendEventStructToIPTask( &xEventMessage, xDontBlock ) != pdPASS )
     {
         /* Failed to send the message, so release the network buffer. */
         vReleaseNetworkBufferAndDescriptor( pxARPWaitingNetworkBuffer );
     }

    /* Clear the buffer. */
    pxARPWaitingNetworkBuffer = NULL;
~~~

Reply:
Oh yes! Nice catch!
Thank you.